File: 1106/twosum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        length = len(nums)
        for i in range(length-1):
            for j in range(i+1,length):
                if target == (nums[i]+nums[j]):
                    return [i,j]

                    
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        length = len(nums)
        for i in range(length-1):
            for j in range(i+1,length):
                if target == (nums[i]+nums[j]):
                    return [i,j]
    

File: 1106/twosum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        for(int i = 0; i < nums.size(); i++){
            int remaining = target - nums[i];

            if(hash.count(remaining)){
                return {hash[remaining], i};
            }
            hash[nums[i]] = i;
        }
    return {0,0};
    }
};